thrust/testing/catch2_test_iterator_traits.cu

```diff
@@ -6,6 +6,43 @@
 
 #include "catch2_test_helper.h"
 
+template <typename Cat, typename Tag>
+inline constexpr bool system_is = cuda::std::is_same_v<thrust::detail::iterator_category_to_system_t<Cat>, Tag>;
+
+TEST_CASE("iterator_category_to_system", "[iterators]")
+{
+  STATIC_REQUIRE(system_is<thrust::input_device_iterator_tag, thrust::device_system_tag>);
+  STATIC_REQUIRE(system_is<thrust::output_device_iterator_tag, thrust::device_system_tag>);
+  STATIC_REQUIRE(system_is<thrust::forward_device_iterator_tag, thrust::device_system_tag>);
+  STATIC_REQUIRE(system_is<thrust::bidirectional_device_iterator_tag, thrust::device_system_tag>);
+  STATIC_REQUIRE(system_is<thrust::random_access_device_iterator_tag, thrust::device_system_tag>);
+
+  STATIC_REQUIRE(system_is<thrust::input_host_iterator_tag, thrust::host_system_tag>);
+  STATIC_REQUIRE(system_is<thrust::output_host_iterator_tag, thrust::host_system_tag>);
+  STATIC_REQUIRE(system_is<thrust::forward_host_iterator_tag, thrust::host_system_tag>);
+  STATIC_REQUIRE(system_is<thrust::bidirectional_host_iterator_tag, thrust::host_system_tag>);
+  STATIC_REQUIRE(system_is<thrust::random_access_host_iterator_tag, thrust::host_system_tag>);
```

Comment:
Ditto, we need to ensure that when deducing the common system we properly filter out bad combinations

Reply:
Could you please be more explicit what you mean? Are you referring to `minimum_system`? And what is a bad combination? `unrelated_systems`?